homeassistant/components/bond/cover.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Bond covers."""
+import asyncio
+import logging
+from typing import Any, Callable, Dict, List, Optional
+
+from bond import BOND_DEVICE_TYPE_MOTORIZED_SHADES, Bond
+
+from homeassistant.components.cover import DEVICE_CLASS_SHADE, CoverEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .utils import BondDevice, get_bond_devices
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity]], None],
+) -> None:
+    """Set up Bond cover devices."""
+
+    bond: Bond = hass.data[DOMAIN][entry.entry_id]
+
+    async def discover():
+        devices = await get_bond_devices(hass, bond)
+        covers = [
+            BondCover(bond, device)
+            for device in devices
+            if device.type == BOND_DEVICE_TYPE_MOTORIZED_SHADES
+        ]
+        async_add_entities(covers)
+
+    asyncio.create_task(discover())
+
+
+class BondCover(CoverEntity):
+    """Representation of a Bond cover."""
+
+    def __init__(self, bond: Bond, device: BondDevice):
+        """Create HA entity representing Bond cover."""
+        self._bond = bond
+        self._device = device
+
+    @property
+    def device_class(self) -> Optional[str]:
+        """Get device class."""
+        return DEVICE_CLASS_SHADE
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Get unique ID for the entity."""
+        return self._device.device_id
+
+    @property
+    def name(self) -> Optional[str]:
+        """Get entity name."""
+        return self._device.name
+
+    @property
+    def device_info(self) -> Optional[Dict[str, Any]]:
+        """Get a an HA device representing this cover."""
+        return {ATTR_NAME: self.name, "identifiers": {(DOMAIN, self.unique_id)}}
```

Comment:
since this is all connected to the hub/bridge device, it might be good to register a hub device and use via_device here, if you can get info about the hub like serial or other unique id and maybe firmware version.

https://developers.home-assistant.io/docs/device_registry_index/#defining-devices

Reply:
There is a get-version API on the hub that returns minimal info, including ID and firmware version. Unfortunately the underling PyPi library (https://pypi.org/project/bond-home/) does not expose it. I submitted a PR to them to add this functionality (https://github.com/nguyer/bond-home/pull/6).
Once it is available I am happy to follow up a PR in HA to utilize via_device relationship

Reply:
It's good to not conflate the device id with the unique_id. The unique_id is a unique id for the entity and in the device info we want to use the unique id for the device. So use `self._device.device_id` here.

Reply:
> since this is all connected to the hub/bridge device, it might be good to register a hub device and use via_device here, if you can get info about the hub like serial or other unique id and maybe firmware version.
> 
> https://developers.home-assistant.io/docs/device_registry_index/#defining-devices

Thanks, @ctalkington - I finally got the upstream library upgraded with hub version endpoints, so when this is reviewed (https://github.com/home-assistant/core/pull/37764) I am going to add Hub device and via_device links

Reply:
> since this is all connected to the hub/bridge device, it might be good to register a hub device and use via_device here, if you can get info about the hub like serial or other unique id and maybe firmware version.
> 
> https://developers.home-assistant.io/docs/device_registry_index/#defining-devices

Ok, here is my attempt on adding hub and via_device: https://github.com/home-assistant/core/pull/37772. Let me know if I on the right path - thanks!